client/src/components/Home.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from '@material-ui/core/Button';
+import Icon from '@material-ui/core/Icon';
+import Grid from '@material-ui/core/Grid';
+import { withStyles } from '@material-ui/core/styles';
+
+const styles = theme => ({
+  button: { margin: theme.spacing.unit },
+  rightIcon: { marginLeft: theme.spacing.unit },
+});
```

Comment:
Great use of theme over here. But how are you using the theme over here? For theme you also have to provide the Provide in root resolver i.e `App` resolver, I suppose.

Reply:
It's the default theme provided by `@material-ui/core/styles`.

I took it from here https://material-ui.com/demos/buttons/

Reply:
In the very last line `export default withStyles(styles)(Home);` I've included hoc `withStyles` which resolves the theme.

Reply:
This is a very simple example of coding by coincidence vs coding deliveretly. How it resolves the value of theme? What is the value of theme? How we can change the theme? Do we need theme over here? And other questions.

Reply:
1. **Do we need theme?** The team has decided to use material UI theme on our slack so that we have a consistent design across the app and we don't have to think much about it. 
2. **How can we change the theme?** We can inject the `MuiThemeProvider` into the root component. 

```
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
import App from './App';

const theme = createMuiTheme({
  spacing: {
    unit: 16,
  },
  typography: {
    fontSize: 28,
  },
});

ReactDOM.render(
  <MuiThemeProvider theme={theme}>
    <App />
  </MuiThemeProvider>,
  document.getElementById('root'),
);


```

However, doing this is optional and the default is theme provided by the npm package. 


3. **How it resolves the value of theme?** I'm not sure about this. But since I'm using the functions provided by them, I assume it is handled by the material-ui package.

Reply:
So yes now if you use the provider you can change the mui theme, anywhere you want, you don't have to depend upon the default settings.